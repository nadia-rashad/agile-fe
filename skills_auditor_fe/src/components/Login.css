h1 {
    font-weight: 550 !important;
}

#root {
    background-color: #dfe8de;
    height: 100vh;
}