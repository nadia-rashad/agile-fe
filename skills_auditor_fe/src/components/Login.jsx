import React, {useEffect} from 'react';
import '../components/styles/styles.css'
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import * as api from '../api';
import { useState } from "react";
import toast, { Toaster } from 'react-hot-toast';
import Home from './Home';


function Login({setToken, setUser}) {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    const onInputEmail = ({target:{value}}) => setEmail(value);
    const onInputPassword = ({target:{value}}) => setPassword(value);
    const [authenticated, setauthenticated] = useState(null);

    const notify = () => toast('Successful Login');

    useEffect(() => {
      const loggedInUserToken = localStorage.getItem("token");
      if (loggedInUserToken) {
        setauthenticated(loggedInUserToken);
      }
    }, []);
   
    const onFormSubmit = async (e) => {
        e.preventDefault()
  
        const credentials = {
          email: email,
          password: password
        }

      await api.checkUserCredentials(credentials).then((res) => {
        setToken(res.data.token);
        setUser(res.data.details);
        localStorage.setItem('token', JSON.stringify(res.data.token) )
        localStorage.setItem('user', JSON.stringify(res.data.details))

        notify();
      
        }
       ).catch(err => {
        toast("Unsuccessful login, please try again")
       })
      }

    return(
        <div className='container'>

          {!authenticated ?  
<>
          <h2>Login</h2>

          
     <Form onSubmit={onFormSubmit}>
      <Form.Group className="mb-3" controlId="formBasicEmail">
        <Form.Label>Email address</Form.Label>
        <Form.Control type="email" placeholder="Enter email" onChange={onInputEmail}/>

      </Form.Group>

      <Form.Group className="mb-3" controlId="formBasicPassword">
        <Form.Label>Password</Form.Label>
        <Form.Control type="password" placeholder="Password" onChange={onInputPassword} />
      </Form.Group>
     
      <Button variant="primary" type="submit">
        Submit
      </Button>
      <Toaster toastOptions={{
    className: '',
    style: {
      border: '2px solid #713200',
      padding: '16px',
      color: '#713200',
    },
  }}/>
    </Form>
    </>
: <Home/>}
       </div>
    )
}

export default Login;