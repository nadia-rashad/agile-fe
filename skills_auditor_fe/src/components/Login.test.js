import renderer from 'react-test-renderer';
import Login from './Login';
import React from 'react';
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
require('jest-localstorage-mock');

describe('Login component testing', () => {
    it('renders the login component', async () => {

        const loginPage = renderer.create(
            <Login/>
        );

        //render( <Login/>)

       let page = loginPage.toJSON();

       await screen.findByRole('heading')

        expect(screen.getByRole('heading')).toHaveTextContent('Login')
        expect(screen.getByRole('button')).toBeDisabled() 
        
    });

});