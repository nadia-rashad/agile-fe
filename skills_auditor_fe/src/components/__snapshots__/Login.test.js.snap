// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component testing renders the login component 1`] = `
<div
  className="container"
>
  <h2>
    Login
  </h2>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "bottom": 16,
          "left": 16,
          "pointerEvents": "none",
          "position": "fixed",
          "right": 16,
          "top": 16,
          "zIndex": 9999,
        }
      }
    />
  </form>
</div>
`;
